scripts/compile/compile-corehost.ps1

```diff
@@ -5,42 +5,163 @@
 
 . $PSScriptRoot\..\common\_common.ps1
 
-header "Building corehost"
-pushd "$RepoRoot\src\corehost"
-try {
-    if (!(Test-Path "cmake\$Rid")) {
-        mkdir "cmake\$Rid" | Out-Null
-    }
-    cd "cmake\$Rid"
-    cmake ..\.. -G "Visual Studio 14 2015 Win64"
-    $pf = $env:ProgramFiles
-    if (Test-Path "env:\ProgramFiles(x86)") {
-        $pf = (cat "env:\ProgramFiles(x86)")
+$CoreHostRelativeDir = "runtimes\\$RID\\native"
+$CoreHostPackageName = "Microsoft.DotNet.CoreHost"
+function main
+{
+    header "Building corehost"
+    pushd "$RepoRoot\src\corehost"
+    try {
+        if (!(Test-Path "cmake\$Rid")) {
+            mkdir "cmake\$Rid" | Out-Null
+        }
+        cd "cmake\$Rid"
+
+        # Pass parameters to cmake through config files.
+        $Utf8Encoding = New-Object System.Text.UTF8Encoding($False)
+        [System.IO.File]::WriteAllLines("$RepoRoot\src\corehost\packaging\.relative", $CoreHostRelativeDir, $Utf8Encoding)
+        [System.IO.File]::WriteAllLines("$RepoRoot\src\corehost\packaging\.name", $CoreHostPackageName, $Utf8Encoding)
+
+        cmake ..\.. -G "Visual Studio 14 2015 Win64"
```

Comment:
We should add a check for CMake executing successfully or not (on the Unix side too).